iOS_SDK/OneSignalSDK/Source/OSInAppMessageTracker.m

```diff
@@ -86,7 +86,7 @@ - (void)initInfluencedTypeFromCache {
 }
 
 - (void)cacheState {
-    [self.dataRepository cacheIAMInfluenceType:self.influenceType];
+    [self.dataRepository cacheIAMInfluenceType:self.influenceType == DIRECT ? INDIRECT : self.influenceType];
 }
```

Comment:
Does this do the same thing as Android `cacheState` method? Doesn't look like you have the `nil` check

Reply:
because the inits between iOS and Android are slightly differents so iOS won't reach the point of having influence type nil, but I will add it to stay consistent between platforms

Reply:
I checked again, and the reason behind this is because on iOS enums cant be null, so it will always default. That why there is no nil check